psychopy/parallel/__init__.py

```diff
@@ -31,22 +31,32 @@
 import sys
 from psychopy import logging
 
-# To make life easier, only try drivers which have a hope in heck of working
+# To make life easier, only try drivers which have a hope in heck of working.
+# Because hasattr() in connection to windll ends up in an OSError trying to
+# load 32bit drivers in a 64bit environment, this passage is written as an
+# inconvenient try-except-construction.
 if sys.platform.startswith('linux'):
     from ._linux import PParallelLinux
     ParallelPort = PParallelLinux
 elif sys.platform == 'win32':
+    drivers = dict(inpout32=['_inpout', 'PParallelInpOut'],
+                   inpoutx64=['_inpout', 'PParallelInpOut'],
+                   dlportio=['_dlportio', 'PParallelDLPortIO'])
     from ctypes import windll
-    if hasattr(windll, 'inpout32'):
-        from ._inpout32 import PParallelInpOut32
-        ParallelPort = PParallelInpOut32
-    elif hasattr(windll, 'dlportio'):
-        from ._dlportio import PParallelDLPortIO
-        ParallelPort = PParallelDLPortIO
-    else:
+    from importlib import import_module
+    for key, val in drivers.items():
+        try:
+            hasattr(windll, key)
+            import_module('.'+val[0], __name__)
+            ParallelPort = getattr(locals()[val[0]], val[1])
```

Comment:
For clarity, I would suggest we do sth. like:

```python
       try:
            hasattr(windll, key)
            driver_name, class_name = val

            import_module('.' + driver_name, __name__)
            ParallelPort = getattr(driver_name, class_name)
```

I also removed the reference to `locals()` -- is this really necessary? (Haven't tested the code)

Reply:
First argument for `getattr()` is the object, and the second is only the name of an object. Therefore, `locals()` is necessary. Or is there another way of calling the object using its name?